app/routes/surveys/components/SubmissionEditor.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+
+import styles from './surveys.css';
+import React, { Component } from 'react';
+import { DetailNavigation, ListNavigation } from '../utils.js';
+import LoadingIndicator from 'app/components/LoadingIndicator';
+import { Field } from 'redux-form';
+import Button from 'app/components/Button';
+import {
+  TextInput,
+  RadioButton,
+  SelectInput,
+  RadioButtonGroup
+} from 'app/components/Form';
+import { createValidator, required } from 'app/utils/validation';
+import { reduxForm } from 'redux-form';
+import type { SurveyEntity } from 'app/reducers/surveys';
+
+type Props = {
+  survey: SurveyEntity,
+  submitting: boolean,
+  handleSubmit: ((SurveyEntity) => Promise<*>) => void,
+  autoFocus: any,
+  fetching: boolean,
+  submitFunction: (SurveyEntity, ?number) => Promise<*>,
+  deleteSurvey: number => Promise<*>
+};
+
+class SubmissionEditor extends Component<Props> {
+  onSubmit = (formContent: Object) => {
+    const { survey, submitFunction } = this.props;
+    return submitFunction({
+      ...formContent,
+      logo: formContent.logo || undefined,
+      studentContact:
+        formContent.studentContact && Number(formContent.studentContact.id),
+      surveyId: survey && survey.id
+    });
+  };
+  render() {
+    const {
+      survey,
+      submitting,
+      autoFocus,
+      handleSubmit,
+      fetching,
+      deleteSurvey
+    } = this.props;
+
+    if (fetching) {
+      return <LoadingIndicator />;
+    }
+
+    const nameField = (
+      <Field
+        placeholder="Tittel"
+        label="Undersøkelse etter Bekk Miniseminar"
+        autoFocus={autoFocus}
+        name="title"
+        component={TextInput.Field}
+        className={styles.editTitle}
+      />
+    );
+
+    return (
+      <div className={styles.root}>
```

Comment:
Should use the `app/components/Content` rather than creating your own container.